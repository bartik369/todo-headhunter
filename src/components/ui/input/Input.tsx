import React, {FC, useState} from 'react';
import { useAppDispatch } from '../../../hooks/useReduxHook';
import { addTodo } from '../../../store/TodoSlice';
import * as todoConst from '../../../constants/todos';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faChevronDown, faSquarePlus} from "@fortawesome/free-solid-svg-icons";
import style from './Input.module.scss';

const Input:FC = () => {
    const [title, setTitle] = useState<string>('');
    const dispatch = useAppDispatch();

    const todoHandler = () => {
        title && dispatch(addTodo(title));
        setTitle('');
    }
    const checkPressEnter = (e:React.KeyboardEvent) => {
        if (e.key === "Enter") todoHandler();
    }
    
    return (
        <div className={style.input}>
            <input
            placeholder={todoConst.inputPlaceholder}
            onChange={(e) => setTitle(e.target.value)}
            onKeyDown={(e) => checkPressEnter(e)}
            value={title}
            type="text"
            />
            <FontAwesomeIcon 
                className={style.icon} 
                icon={faChevronDown}
            />
            {title &&
            <button name="add-btn" onClick={todoHandler} className={style.add}>
                <FontAwesomeIcon title={todoConst.addTodo} icon={faSquarePlus}/>
            </button>
            }
        </div>
    );
};

export default Input;