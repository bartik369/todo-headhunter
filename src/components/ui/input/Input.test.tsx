import { render} from "@testing-library/react";
import Input from "./Input";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store'


describe('Test input component', () => {
    const initialState = {
        todos: [],
    }
    const mockStore = configureStore();
    let store = mockStore(initialState);

    test('Render', () => {
        render (
            <Provider store={store}>
                 <Input />
            </Provider>
        )
    });

    test('Snapshot', () => {
        const input = render (
            <Provider store={store}>
                 <Input />
            </Provider>
        )
        expect(input).toMatchSnapshot();
    });
});