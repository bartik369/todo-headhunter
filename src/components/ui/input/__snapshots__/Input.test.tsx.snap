// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test input component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="input"
      >
        <input
          placeholder="What needs to be done?"
          type="text"
          value=""
        />
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down icon"
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="input"
    >
      <input
        placeholder="What needs to be done?"
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-chevron-down icon"
        data-icon="chevron-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
